src/vision/status_quo/alan_picks_web_server.md

```diff
@@ -0,0 +1,72 @@
+
+The [arewewebyet site](https://www.arewewebyet.org/) mentions four good web servers. Alan picks Tide because the interfaces and the emphasis on middleware reminds him of Nodejs' Express framework.
+
+
+### The first endpoint
+Alan sets up all the boilerplate and is ready to write the first endpoint. He picks `PUT /support-ticket` because it barely has any logic in it. When a request arrives, the handler only makes a request to Zendesk to create a support ticket. The handler is stateless and has no middleware.
+
+The [arewewebyet site](https://www.arewewebyet.org/) doesn't recommend a specific http client, so Alan searches for one in crates.io. He picks reqwest simply because it's the most popular.
+
+Alan combines the knowledge he has from programming in synchronous Rust and asynchronous Javascript to come up with a few lines that should work. If the compiler is happy, then so is he!
+
+### First problem: incompatible runtimes
+
+The first problem he runs into is very similar to the one described in [the compiler trust story](https://github.com/rust-lang/wg-async-foundations/blob/master/src/vision/status_quo/alan_started_trusting_the_rust_compiler_but_then_async.md#fractured-futures-fractured-trust).
+
+In short, Alan has problems because Tide is based on `std-async` and reqwest on the latest version of `tokio`. This is a real pain for Alan as he has now to change either the http client or the server so that they use the same runtime.
+
+He decides to switch to Actix web.
+
+### Second problem: incompatible versions of the same runtime
+
+Alan does all the changes and again the compiler seems to be happy. To his surprise, the same problem happens again. The program panics with the message `there is no reactor running, must be called from the context of a Tokio 1.x runtime`. He is utterly puzzled as Actix web is based on Tokio just like reqwest.
+
+It turns out that the issue is that Alan's using v0.11.2 of reqwest, which uses tokio v1, and v3.3.2 of actix-web, which uses tokio v0.3.
+
+The solution to this problem is then to dig into all the versions of `reqwest` until he finds one which uses the same version of tokio.
+
+### Can Alan sell the Rust migration to his boss?
+
+This experience has made Alan think twice about whether Rust is indeed web ready. On the one hand, there are very good libraries for web servers, ORMs, parsers, session management, etc. On the other, Alan is fearful that in 2/3/6 months time he has to develop new features with libraries that already exist but turn out to be incompatible with the runtime chosen at the beginning of the project.
+
+## 🤔 Frequently Asked Questions
+
+### **What are the morals of the story?**
+* Rust's ecosystem is ready for the web. There are a ton of brilliant libraries and frameworks out there. But web developers that are new to Rust may struggle with the idea that before picking a web server, they have to pick an async runtime.
```

Comment:
```suggestion
* Rust's ecosystem has a lot of great components that may individually be ready for the web, but combining them is still a fraught proposition. It can also be confusing to web developers that are new to Rust, who may struggle with the idea that before picking a web server, they have to pick an async runtime.
```

I feel like this moral is a bit too positive as phrased :)

Reply:
Excellent phrasing! I still love Rust too much